### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Bajkowsky**

### CODE REVIEW

File: src/arrayReverse.test.js [hunk 1]
```diff
@@ -8,13 +8,29 @@ describe(`Function 'arrayReverse':`, () => {
...
+  });
+
+  it(`should return an inverted array preserving the length`
+    + `and order of the elements of the original array`, () => {
     expect(arrayReverse(['Mate', 'Academy'])).toEqual(['ymed', 'acAetaM']);
   });
 
-  // write more tests here
+  it(`should return an inverted array preserving the length`
+    + `and order of the elements of the original array`, () => {
```
Comment (natalia-klonowska) [inline src/arrayReverse.test.js:32]: you should change this to not exactly repeat previous test. you can mention using special symbols

**Review by natalia-klonowska — CHANGES_REQUESTED** [review]
you should add more tests. check empty array and array with numbers

**Review by natalia-klonowska — CHANGES_REQUESTED** [review]
you should add more tests. check empty array and array with numbers

File: src/arrayReverse.test.js (hunk 1 above)
Comment (Bajkowsky) [reply src/arrayReverse.test.js]: what if not all tests pass?

Comment (Bajkowsky) [reply src/arrayReverse.test.js]: i must change the function ?

Comment (Bajkowsky) [reply src/arrayReverse.test.js]: √ should be declared (5ms)
    √ should return an array
    √ should return an empty array if the original array is empty (1ms)
    × should return an inverted array consisting of numbers (5ms)
    × should return an array where each element with special characters is reversed (1ms)
    × should return an inverted string array with varying lengths (1ms)
    √ should return an inverted array preserving the length and order of elements including special characters
    × should return an inverted array while handling mixed data types (1ms)

**Review by Zibi95 — CHANGES_REQUESTED** [review]
There are missing some tests for example:
- Test for an empty array input
- Test for non-string elements
- Test for undefined or null input

**Review by natalia-klonowska — CHANGES_REQUESTED** [review]
you didn't add any new commit with additional tests that were mentioned in last review

**Comment by Bajkowsky** [comment]
> I'm confused, there were additional tests with other types of elements and after last commit you removed some of them.

because this tests not passed and i cant push the code

    √ should be declared (2ms)
    √ should return an array
    √ should return an empty array when the input array is empty (1ms)
    √ should return an array with empty strings (1ms)
    √ should reverse a single word
    √ should handle an array with special characters (1ms)
    √ should handle an array with 1 special symbol
    √ should return an array with reversed strings of varying lengths
    √ should handle an array with identical strings (1ms)
    √ should return an empty array when the input is an empty array
    × should handle non-string and non-reversible elements without changes (5ms)
    × should reverse elements with spaces while keeping spaces in the right positions (1ms)

**Comment by Bajkowsky** [comment]
to fix this i must change the function arrayReverse.js then tests are passed

**Review by Zibi95 — CHANGES_REQUESTED** [review]
You shouldn't change the implementation of the function. Instead, you should write tests that ensure it behaves as expected. The function should convert a number element to a string, and in the case of undefined or null, it should throw an error.

**Review by Zibi95 — CHANGES_REQUESTED** [review]
Alright, we're back to square one. Now, please add additional tests for numbers and empty arrays.

**Review by danon321 — APPROVED** [review]
